openformsclient/mixins.py

```diff
@@ -0,0 +1,16 @@
+from .models import OpenFormsSlugField, OpenFormsUUIDField
+
+
+class OpenFormsClientMixin:
```

Comment:
@pi-sigma I'm not really a fan of having to use a mixin on a `ModelAdmin` to get these choices to work. @sergei-maertens what do you think?

Isn't this possible in another why, perhaps by making `get_choices` lazy?

Reply:
I tested this locally and it seems to work:

```python
class OpenFormsBaseField:
    ...

    def get_choices(
        self,
        include_blank=True,
        blank_choice=BLANK_CHOICE_DASH,
        limit_choices_to=None,
        ordering=(),
    ):
        from django.utils.functional import lazy

        def _fetch():
            cache_key = f"openformsclient.models.OpenFormsFieldMixin.get_choices__use_uuids_{self.use_uuids}"

            choices = cache.get(cache_key)
            if choices is None:
                try:
                    choices = get_form_choices(use_uuids=self.use_uuids)
                except Exception as e:
                    logger.exception(e)
                    choices = []
                else:
                    cache.set(cache_key, choices, timeout=60)

            if choices:
                if include_blank:
                    blank_defined = any(choice in ("", None) for choice, _ in choices)
                    if not blank_defined:
                        choices = blank_choice + choices
            return choices
        return lazy(_fetch, list)
```

This also removes the need to have the change made in https://github.com/open-formulieren/open-forms-client-django/commit/6c79b554b8a926d37db5775578ce46aed05a992d, so that can be removed as well I think

Reply:
Nice, this works indeed. Is there a reason for choosing `OpenFormsFieldMixin` in the `cache_key` as opposed to `OpenFormsBaseField`?

Reply:
Ah no, that was a mistake, you can just use `OpenFormsBaseField` there instead